### PR TITLE
Not accurate error message for invalid authenticated requests

### LINKED ISSUE #850: PostObject: not accurate error message for invalid authenticated requests
If authenticated PostObject request is made with invalid policy form (when x-amz-credential is not provided), s3 gw returns this error, that is very misleading, it contains some hardcoded info about the region:

```
<?xml version="1.0" encoding="UTF-8"?>
<Error><Code>AuthorizationHeaderMalformed</Code><Message>The authorization header is malformed; the region is wrong; expecting &#39;us-east-1&#39;.</Message><BucketName>yournamehere-mb87trhcltvq2dec-1</BucketName><Resource>/yournamehere-mb87trhcltvq2dec-1</Resource><RequestId>e1c32bd2-7f1e-47b9-8083-de6dd9be3edf</RequestId><HostId>a3215bbd-8b99-4596-9f1d-00a42618ab1c</HostId></Error>
```

More precise error message would be like the one from MINIO:

```
<?xml version="1.0" encoding="UTF-8"?>
<Error><Code>AuthorizationQueryParametersError</Code><Message>Error parsing the X-Amz-Credential parameter; the Credential is mal-formed; expecting &#34;&lt;YOUR-AKID&gt;/YYYYMMDD/REGION/SERVICE/aws4_request&#34;.</Message><BucketName>yournamehere-ns24t1094wwfebma-1</BucketName><Resource>/yournamehere-ns24t1094wwfebma-1</Resource><RequestId>178DD0B3BB5250D0</RequestId><HostId>f25fd5b4-787e-4670-97d2-f1bed82c70ab</HostId></Error>
```

s3 gw log:
```
2023-10-14T17:20:31.494Z        info    api/router.go:162       call method     {"status": 200, "host": "s3.neofs.devenv:8080", "request_id": "6c33bc78-e008-4389-ab9a-f2fb805099e3", "method": "CreateBucket", "bucket": "yournamehere-8k0rni0vdav33wvw-1", "object": "", "description": "OK"}
2023-10-14T17:20:31.520Z        error   api/user_auth.go:39     failed to pass authentication   {"error": "AuthorizationHeaderMalformed: 400 => The authorization header is malformed; the region is wrong; expecting 'us-east-1'."}
```

Same error appears and on another requests, e.g. PutBucketPolicy, PutObject

### PR DESCRIPTION
**Pull request by smallhive**
Closes #850

### CODE REVIEW

File: api/auth/center.go [hunk 1]
```diff
@@ -236,9 +236,9 @@ func (c *center) checkFormData(r *http.Request) (*Box, error) {
 		return nil, ErrNoAuthorizationHeader
 	}
 
-	submatches := c.postReg.GetSubmatches(MultipartFormValue(r, "x-amz-credential"))
+	submatches := c.postReg.GetSubmatches(MultipartFormValue(r, strings.ToLower(AmzCredential)))
```
Comment (roman-khimov) [inline api/auth/center.go:239]: Not really related to 61264b05f52ed6847d5bdb5f210982ae70424bce

Comment (smallhive) [reply api/auth/center.go:239]: Moved to another commit

Comment (roman-khimov) [reply api/auth/center.go:239]: I'd squash it with ee701c4d0762dbc7a1642e93cdaa3b23a9c8a21f.

Comment (smallhive) [reply api/auth/center.go:239]: Done

**Review by roman-khimov — APPROVED** [review]

**Pull request merged**